src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -204,8 +204,9 @@ std::shared_ptr<AbstractOperator> LQPTranslator::_translate_predicate_node_to_in
 
   // All chunks that have an index on column_ids are handled by an IndexScan. All other chunks are handled by
   // TableScan(s).
-  auto index_scan = std::make_shared<IndexScan>(input_operator, SegmentIndexType::GroupKey, column_ids,
-                                                predicate->predicate_condition, right_values, right_values2);
+  auto index_scan =
+      std::make_shared<IndexScan>(input_operator, SegmentIndexType::GroupKey, column_ids,
+                                  predicate->predicate_condition, right_values, right_values2, node->deep_copy());
```

Comment:
Does this have to be part of the constructor? It really has nothing to do with the purpose of the object being constructed.

As the LQPTranslator is the only one setting it, maybe a single virtual call in translate_node would even be easier?

Reply:
Good point. Probably opening a new PR and closing this one to handle that.